src/test/gc.test.ts

```diff
@@ -0,0 +1,143 @@
+      numLiveExecutors--;
+    });
+    registerExecutorConstructorHook((executor) => {
+      numLiveExecutors++;
+      executorFinalizationRegistry.register(executor, null);
+    });
+
+    const executionFinalizationRegistry = new FinalizationRegistry(() => {
+      numLiveExecutions--;
+    });
+    registerExecutionConstructorHook((execution) => {
+      numLiveExecutions++;
+      executionFinalizationRegistry.register(execution, null);
+    });
+    ctx.rig = new WireitTestRig();
+    await ctx.rig.setup();
+  } catch (error) {
+    // Uvu has a bug where it silently ignores failures in before and after,
+    // see https://github.com/lukeed/uvu/issues/191.
+    console.error('uvu before error', error);
+    process.exit(1);
+  }
+});
+
+test.after.each(async (ctx) => {
+  try {
+    numLiveExecutors = 0;
+    numLiveExecutions = 0;
+    await ctx.rig.cleanup();
+  } catch (error) {
+    // Uvu has a bug where it silently ignores failures in before and after,
+    // see https://github.com/lukeed/uvu/issues/191.
+    console.error('uvu after error', error);
+    process.exit(1);
+  }
+});
+
+async function retryWithGcUntilCallbackDoesNotThrow(
+  cb: () => void
+): Promise<void> {
+  for (const wait of [0, 10, 100, 500, 1000]) {
+    global.gc();
+    try {
+      cb();
+      return;
+    } catch {
+      // Ignore
+    }
+    await new Promise((resolve) => setTimeout(resolve, wait));
+  }
+  cb();
+}
+
+test(
+  'standard garbage collection',
+  timeout(async ({rig}) => {
+    const standard = await rig.newCommand();
+    await rig.writeAtomic({
+      'package.json': {
+        scripts: {
+          standard: 'wireit',
+        },
+        wireit: {
+          standard: {
+            command: standard.command,
+          },
+        },
+      },
+    });
+
+    const logger = new DefaultLogger(rig.temp);
+    const script = await new Analyzer().analyze(
+      {packageDir: rig.temp, name: 'standard'},
+      []
+    );
+    if (!script.config.ok) {
+      for (const error of script.config.error) {
+        logger.log(error);
+      }
+      throw new Error(`Analysis error`);
+    }
+
+    const workerPool = new WorkerPool(Infinity);
+    const abort = new Deferred<void>();
+
+    const numIterations = 10;
+    for (let i = 0; i < numIterations; i++) {
+      const executor = new Executor(
+        script.config.value,
+        logger,
+        workerPool,
+        undefined,
+        'no-new',
+        abort,
+        undefined
+      );
+      const resultPromise = executor.execute();
+      assert.ok(numLiveExecutors >= 1);
+      assert.ok(numLiveExecutions >= 1);
+      (await standard.nextInvocation()).exit(0);
+      const result = await resultPromise;
+      if (!result.ok) {
+        for (const error of result.error) {
+          logger.log(error);
+        }
+        throw new Error(`Execution error`);
+      }
+    }
+
+    await retryWithGcUntilCallbackDoesNotThrow(() => {
```

Comment:
how often does this need all the retries to pass?

Reply:
In a single run I just did, one time needed `10`, another needed `100`.